docs/rules/object-key-newlines.md

```diff
@@ -0,0 +1,68 @@
+# Enforce placing object keys on separate lines (object-key-newlines)
+
+While formatting preferences are very personal, a number of style guides require that
+object keys be placed on separate lines for better readability.
+
+## Rule Details
+
+This rule aims to maintain consistency of newlines between object keys.
+
+Examples of **correct** code for this rule:
+
+```js
+/*eslint object-key-newlines: "error"*/
+
+var obj = {
+    foo: "foo",
+    bar: "bar",
+    baz: "baz"
+};
+```
+
+Examples of **incorrect** code for this rule:
+
+```js
+/*eslint object-key-newlines: "error"*/
+
+var obj = { foo: "foo", bar: "bar", baz: "baz" };
+
+var obj2 = {
+    foo: "foo", bar: "bar", baz: "baz"
+};
+
+var obj3 = {
+    foo: "foo", bar: "bar",
+    baz: "baz"
+};
+```
+
+## Options
+
+This rule has an object option:
+
+* `"allowMultipleKeysPerLine"`: `true` allows all keys and values to be on the same line
+
+Examples of additional **correct** code for this rule with the `{ "allowMultipleKeysPerLine": true }` option:
+
+```js
+/*eslint object-key-newlines: ["error", { "allowMultipleKeysPerLine": true }]*/
+
+var obj = { foo: "foo", bar: "bar", baz: "baz" };
+
+var obj2 = {
+    foo: "foo", bar: "bar", baz: "baz"
+};
+```
+## When Not To Use It
+
+You can turn this rule off if you are not concerned with the consistency of newlines between object keys.
+
+## Compatibility
+
+* **JSCS**: `requireObjectKeysOnNewLine`
+
+## Related Rules
```

Comment:
False alarm. Just read the docs and tested some example code to make sure that:
* key-spacing `"never"` ignores properties on multiple lines
* object-curly-spacing `"align"` ignores properties on the same line